### ISSUE TITLE
Exception when calling toIterable without parameters

### ISSUE THREAD

**Issue by flack** [doctrine/doctrine2#8575]
I'm calling Doctrine\ORM\Query::toIterable() without any parameters like this:

```php
$resultset = $query->toIterable();
```

This produces the following exception:

```
TypeError: Argument 1 passed to Doctrine\ORM\Internal\Hydration\AbstractHydrator::toIterable() must implement interface Doctrine\DBAL\Driver\Statement, instance of Doctrine\DBAL\ForwardCompatibility\Result given, called in vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php on line 1010
```

The code used to work fine with the `iterate()` method, but I wanted to replace it with the new version, since it's deprecated. If I downgrade `doctrine/dbal` to 2.12.1, everything works. With 2.13.0 it breaks. Not sure if this means I should rather open a ticket for doctrine/dbal?

**Comment by derrabus** [comment]
I have the same issue here. This is a blocker for upgrading to DBAL 2.13 for us.

**Comment by beberlei** [comment]
Fixed with https://github.com/doctrine/orm/pull/8495 already, not released yet - will be in 2.8.3

**Comment by flack** [comment]
@beberlei awesome, thanks for the quick fix!
